stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -1000,11 +1049,12 @@ impl NakamotoChainState {
         clarity_instance: &'a mut ClarityInstance,
         sortition_dbconn: &'b dyn SortitionDBRef,
         pox_constants: &PoxConstants,
-        chain_tip: &StacksHeaderInfo,
```

Comment:
Can you help me understand why replacing `chain_tip` with `parent_stacks_height` and `parent_burn_height` is required?  This seems needless to me.

Reply:
It reduces the amount that needs to be mocked: the `chain_tip` argument is _only_ used for obtaining the parent stacks height and parent burn height (for epoch checks, mostly). It also makes writing unit tests easier. An argument like `chain_tip` which includes a bunch of information that the function _doesn't_ use confuses the interface.